rc/windowing/detection.kak

```diff
@@ -23,7 +23,7 @@ declare-option -docstring \
 "Ordered list of windowing modules to try and load. An empty list disables
 both automatic module loading and environment detection, enabling complete
 manual control of the module loading." \
-str-list windowing_modules 'tmux' 'screen' 'kitty' 'iterm' 'x11'
+str-list windowing_modules 'tmux' 'screen' 'kitty' 'iterm' 'x11' 'wayland'
```

Comment:
I think we might want to move wayland in front of x11, as Xwayland will likely make loading the x11 module succeed.

Reply:
I'm not sure. This will disallow xwayland compatible terminals completly right ?

Reply:
I dont have wayland here to test (nvidia...) but I would expect the x11 module to load fine thanks to DISPLAY being set which would then prevent loading of the wayland module.

I guess this is a deeper issue that might ultimately warrant a more complex solution (for example, it seems here wayland-terminal and x11-terminal could be shared), I imagine this could be fixed later on.

Reply:
We could still add an xwayland module to permit that. I dont see the point at the moment